### PR TITLE
[12.0][IMP] analytic_tag_dimension: Sync dimension code and description changes

### PR DESCRIPTION
**Pull request by carlosdauden**
Syncronice changes in exsiting dimensions.
Allow extend models.

@Tecnativa

### CODE REVIEW

File: analytic_tag_dimension/models/analytic.py [hunk 1]
```diff
@@ -18,28 +19,51 @@ class AccountAnalyticDimension(models.Model):
...
-        )
+        ]
+
+    def get_field_name(self, code=False):
+        return 'x_dimension_{}'.format(code or self.code)
+
+    @api.model
+    def create(self, values):
+        if ' ' in values.get('code'):
+            raise ValidationError(_("Code can't contain spaces!"))
```
Comment (pedrobaeza) [inline analytic_tag_dimension/models/analytic.py]: Why not putting this as `api.constrains`?

**Review by pedrobaeza — COMMENTED** [review]
Now that I'm seeing the code, I think it would be desirable to inject the fields the same way we do in `base_search_field_filter`, don't you think?

File: analytic_tag_dimension/models/analytic.py [hunk 2]
```diff
@@ -18,28 +19,51 @@ class AccountAnalyticDimension(models.Model):
         string='Analytic Tags')
 
     @api.model
-    def create(self, values):
-        if ' ' in values.get('code'):
-            raise ValidationError(_("Code can't contain spaces!"))
-        model_names = (
+    def get_model_names(self):
```
Comment (kittiu) [inline analytic_tag_dimension/models/analytic.py:28]: Good enhancement. I have one question. As this module is used to extend dimension in other module, i..e, sale.order.line, purchase.order.line
https://github.com/ecosoft-odoo/account-analytic/blob/12.0-mig-analytic_tag_dimension_sale_warning/analytic_tag_dimension_sale_warning

Will this PR has affect on those?

**Review by rafaelbn — APPROVED** [review]
Functionally tested :+1: 

Thanks

**Review by angelmoya — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@carlosdauden what do you think about my proposal?

**Comment by carlosdauden** [comment]
@pedrobaeza I don't know if I understand what you mean.
If it is to create virtual fields to be able to filter, but without the possibility of grouping, I think that precisely the main objective of this module is the possibility of grouping in the pivot views.

**Comment by pedrobaeza** [comment]
Well, I mean to not require the ir.model.fields for doing all (filtering, grouping), injecting them on the fly and relying in a computed field. Is it possible?

**Comment by carlosdauden** [comment]
I don't know

File: analytic_tag_dimension/models/analytic.py (hunk 2 above)
Comment (carlosdauden) [reply analytic_tag_dimension/models/analytic.py:28]: I think that without this PR, the linked module don't have a inheritable way to create dimension columns in mentioned models, if you need.
So this PR beneficially affects if you extend `get_model_names` and does not affect at all if you don't.

**Comment by pedrobaeza** [comment]
OK, let's go as is and let the possible implementation for further investigations:

/ocabot merge minor

**Pull request merged**